cookbook/controller/upload_file.rst

```diff
@@ -0,0 +1,161 @@
+    operations (such as file renaming, saving and deleting) and it's tightly
+    integrated with Doctrine ORM, MongoDB ODM, PHPCR ODM and Propel.
+
+Imagine that you have a ``Product`` entity in your application and you want to
+add a PDF brochure for each product. To do so, add a new property called ``brochure``
+in the ``Product`` entity::
+
+    // src/AppBundle/Entity/Product.php
+    namespace AppBundle\Entity;
+
+    use Doctrine\ORM\Mapping as ORM;
+    use Symfony\Component\Validator\Constraints as Assert;
+
+    class Product
+    {
+        // ...
+
+        /**
+         * @ORM\Column(type="string")
+         *
+         * @Assert\NotBlank(message="Please, upload the product brochure as a PDF file.")
+         * @Assert\File(mimeTypes={ "application/pdf" })
+         */
+        private $brochure;
+
+        public function getBrochure()
+        {
+            return $this->brochure;
+        }
+
+        public function setBrochure($brochure)
+        {
+            $this->brochure = $brochure;
+
+            return $this;
+        }
+    }
+
+Note that the type of the ``brochure`` column is ``string`` instead of ``binary``
+or ``blob`` because it just stores the PDF file name instead of the file contents.
+
+Then, add a new ``brochure`` field to the form that manages ``Product`` entities::
+
+    // src/AppBundle/Form/ProductType.php
+    namespace AppBundle\Form;
+
+    use Symfony\Component\Form\AbstractType;
+    use Symfony\Component\Form\FormBuilderInterface;
+
+    class ProductType extends AbstractType
+    {
+        public function buildForm(FormBuilderInterface $builder, array $options)
+        {
+            $builder
+                // ...
+                ->add('brochure', 'file', array('label' => 'Brochure (PDF file)'))
+                // ...
+            ;
+        }
+
+        // ...
+    }
+
+Now, update the template that renders the form to display the new ``brochure``
+field (the exact template code to add depends on the method used by your application
+to :doc:`customize form rendering </cookbook/form/form_customization>`):
+
+.. code-block:: html+jinja
+
+    {# app/Resources/views/product/new.html.twig #}
+    <h1>Adding a new product</h1>
+
+    {{ form_start() }}
+        {# ... #}
+
+        {{ form_row(form.brochure) }}
+    {{ form_end() }}
+
+Finally, you need to update the code of the controller that handles the form::
+
+    // src/AppBundle/Controller/ProductController.php
+    namespace AppBundle\ProductController;
+
+    use Symfony\Bundle\FrameworkBundle\Controller\Controller;
+    use Symfony\Component\HttpFoundation\Request;
+    use Sensio\Bundle\FrameworkExtraBundle\Configuration\Route;
+    use AppBundle\Entity\Product;
+
+    class ProductController extends Controller
+    {
+        /**
+         * @Route("/product/new", name="app_product_new")
+         */
+        public function newAction(Request $request)
+        {
+            //...
```

Comment:
We might as well show the `$form = $this->createForm()` part - then this entry is totally complete